internal/test/try/try.go

```diff
@@ -36,7 +36,7 @@ func Eventually(fn func() error, duration time.Duration, tick time.Duration) err
 			}
 			return lastErr
 		case <-ticker:
-			lastErr := fn()
+			lastErr = fn()
```

Comment:
was causing the underlying error to not get propagated. this wasn't caught as the tests dont test for the error value, just if an error exists or not. fixed that too